src/main/java/seedu/address/logic/commands/ViewHistoryCommand.java

```diff
@@ -0,0 +1,95 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Id;
+import seedu.address.model.person.Patient;
+import seedu.address.model.person.Person;
+
+/**
+ * Views the medical history of an existing patient in the system.
+ */
+public class ViewHistoryCommand extends Command {
+
+    public static final String COMMAND_WORD = "viewHistory";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Views the medical history of the patient identified "
+            + "by the patient ID. "
+            + "Parameters: PATIENT_ID (must be a valid ID) "
+            + "LOCAL_DATETIME \n"
+            + "Example: " + COMMAND_WORD + " P1234567A 2023-09-25T10:15";
+
+    public static final String MESSAGE_VIEW_HISTORY_SUCCESS = "Viewed history for Patient: %1$s on %2$s";
+    public static final String MESSAGE_NO_HISTORY_FOUND = "No history found for Patient: %1$s on %2$s";
+
+    private final Id patientId;
+    private final LocalDateTime dateTime;
+
+    /**
+     * @param patientId of the patient to view the history of
+     * @param dateTime the specific date and time of the history to view (optional)
+     */
+    public ViewHistoryCommand(Id patientId, LocalDateTime dateTime) {
+        requireNonNull(patientId); // Only patientId is mandatory
+        this.patientId = patientId;
+        this.dateTime = dateTime;
+    }
+
+    /**
+     * @param patientId of the patient to view the history of
+     */
+    public ViewHistoryCommand(Id patientId, Id doctorId) {
```

Comment:
Noticed that doctorId is a parameter in the constructor, but it isn't used anywhere in the method body.